cmd/serve.go

```diff
@@ -127,6 +131,17 @@ func startServer(logger log.Logger, appConfig *config.Shield, err error, ctx con
 	return s
 }
 
+func customHeaderMatcherFunc(proxyHeader string) func(key string) (string, bool) {
```

Comment:
How about making header matcher for `[]string` and not just for 1 key. Have a look at how [DefaultHeaderMatcher](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/mux.go#L107) is implemented